README.md

```diff
@@ -1,55 +1,142 @@
 
-## Prerequisites
+## Setup to use Docker
+
+The example code from this repository can run in Docker containers with default  configurations. The Docker containers use LocalStack to run a localized environment that simulates interactions with AWS. This is the quickest way to spin up a working development environment (without doing additional configuration.) 
+
+NOTE: This is a development setup and should not be deployed as-is to a production environment.
+
+### Supported Docker versions
+
+The example runs with Docker Desktop version 4.34.3 (170107) or later.
+
+### Run the local setup
+
+Build and run the Docker containers by entering this command:
+
+```shell
+make local
+```
+
+The command will first build and then run the containers using the [docker-compose file](docker-compose.yaml), which does the following:
+
+1. Uses LocalStack to run a mock AWS infrastructure in a container called `localstack-main`.
+2. The setup scripts will run (using the code from [setup.py](setup.py) and [local-setup.sh](local-setup.sh)) from a container called `local-setup` and configure the example, automating previous manual steps to have the supporting infrastructure (for example, creating mocked AWS users in LocalStack and certificate infrastructure). The container exits once the setup is done. The files created during setup are copied for reference to the `config_volume` directory in the root of this repository. Note that changing the values of the configurations in this directory does not affect the running Docker setup, nor after restarting it.
+3. The signing server starts with default configuration in the `local-signer` container.
+4. Once the signing server is ready, the example runs a self-check using the Python client and verifies that a default image placed in `client_volume/signed-images` can be signed using a `local-client` container. You can then also see the signed test file in a directory created at the root of this repo, `client_volume/signed_images`.
+
+When the `make local` command finishes, it displays something like this:
+
+```shell
+(...Docker images build details...)
+
+--- Running containers.........
+
+docker compose up -d
+[+] Running 5/5
+ ✔ Network c2pa-python-example_default  Created                                                              0.0s
+ ✔ Container localstack-main            Started                                                              0.3s
+ ✔ Container local-setup                Exited                                                               0.8s
+ ✔ Container local-signer               Healthy                                                              6.4s
+ ✔ Container local-client               Started
+```
+
+### Re-run the Python client container
```

Comment:
It might be helpful to remind the reader that`make local` runs the client once, which is why this is "re-running" it.

Reply:
Done.